### ISSUE TITLE
Replace the "configured" uuid dependency with a trivial system version. This is the version we were always using anyway, regardless of the installation of OSSP-UUID [phab:D2174]

### ISSUE THREAD

**Issue by kythe** [google/kythe#2636]
_Created by **shahms** at 2018-02-08 18:01:14_: 

On Linux, we were typically using the system version found via pkg-config. On macOS, we were finding the OSSP version which was adding the correct -luuid, but was never itself being used.  Take out the convoluted and complicated middleware.

**Comment by kythe** [comment]
_**salguarnieri** wrote at 2018-02-08 18:18:03_:

Thanks for digging this up.

I get the following error with a build.

I did: `bazel clean --expunge` then `./tools/modules/update.sh`, before I ran `bazel build //...`

```
ERROR: /Users/salguarnieri/Groksrc/kythe/kythe/cxx/extractor/testdata/BUILD:254:1: CcIndexCompilation kythe/cxx/extractor/testdata/extract_verify_iostream_test_entries/extract_verify_iostream_test.cc.kindex.entries failed (Exit 1)
/private/var/tmp/_bazel_salguarnieri/ca16ea5a435d8ac2288f379cdf092007/bazel-sandbox/4678205669226231124/execroot/io_kythe/kythe/cxx/extractor/testdata/extract_verify_iostream_test.cc:1:10: fatal error: 'iostream' file not found
#include <iostream>
         ^~~~~~~~~~
1 error generated.
```

**Comment by kythe** [comment]
_**shahms** wrote at 2018-02-08 18:19:29_:

>>! In (earlier comment), salguarnieri wrote:
> Thanks for digging this up.
> 
> I get the following error with a build.
> 
> I did: `bazel clean --expunge` then `./tools/modules/update.sh`, before I ran `bazel build //...`
> 
> ```
> ERROR: /Users/salguarnieri/Groksrc/kythe/kythe/cxx/extractor/testdata/BUILD:254:1: CcIndexCompilation kythe/cxx/extractor/testdata/extract_verify_iostream_test_entries/extract_verify_iostream_test.cc.kindex.entries failed (Exit 1)
> /private/var/tmp/_bazel_salguarnieri/ca16ea5a435d8ac2288f379cdf092007/bazel-sandbox/4678205669226231124/execroot/io_kythe/kythe/cxx/extractor/testdata/extract_verify_iostream_test.cc:1:10: fatal error: 'iostream' file not found
> #include <iostream>
>          ^~~~~~~~~~
> 1 error generated.
> ```

That is unrelated to this change and is fallout from (earlier comment).  I have a partial reversion of that which I can send first, though.

**Comment by kythe** [comment]
_**salguarnieri** wrote at 2018-02-08 18:25:06_:

SGTM.

I think you linked the wrong diff in the previous post, but I know what you are referring to.

**Comment by kythe** [comment]
_**salguarnieri** accepted this revision at 2018-02-08 18:25:06_

**Comment by kythe** [comment]
_**shahms** wrote at 2018-02-08 18:28:05_:

Ooops, yeah, the offending diff was #2548

**Comment by kythe** [comment]
_**shahms** closed via commit b4ff29ba9fd536e89c531fa002145c29b1c7cddf at 2018-02-08 21:20:22_
